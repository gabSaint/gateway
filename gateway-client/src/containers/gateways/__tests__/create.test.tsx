import React from "react";
import { ReactWrapper, shallow, ShallowWrapper } from "enzyme";
import renderer from "react-test-renderer";

import CreateGateway from "../create";

describe("CreateGateways", () => {
  let wrapperCreateGateway: ReactWrapper | ShallowWrapper;

  beforeEach(() => {
    wrapperCreateGateway = shallow(<CreateGateway />);
  });

  it("renders correctly", () => {
    const tree = renderer.create(<CreateGateway />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders title", () => {
    const title = wrapperCreateGateway.find("h2");
    expect(title).toHaveLength(1);
  });

  it("renders form", () => {
    const form = wrapperCreateGateway.find("GatewayForm");
    expect(form).toHaveLength(1);
  });
});
