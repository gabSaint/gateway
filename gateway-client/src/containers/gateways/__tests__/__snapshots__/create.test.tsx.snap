// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateGateways renders correctly 1`] = `
Array [
  <h2>
    New Gateway
  </h2>,
  <form
    action="#"
    className="ui form"
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <div
      className="field"
    >
      <label
        htmlFor="input-serial"
      >
        Serial Number
      </label>
      <input
        id="input-serial"
        name="serial"
        onBlur={[Function]}
        onChange={[Function]}
        value=""
      />
    </div>
    <div
      className="field"
    >
      <label
        htmlFor="input-name"
      >
        Name
      </label>
      <input
        id="input-name"
        name="name"
        onBlur={[Function]}
        onChange={[Function]}
        value=""
      />
    </div>
    <div
      className="field"
    >
      <label
        htmlFor="input-address"
      >
        IPv4 Address
      </label>
      <input
        id="input-address"
        name="address"
        onBlur={[Function]}
        onChange={[Function]}
        value=""
      />
    </div>
    <button
      type="submit"
    >
      Submit
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      Cancel
    </button>
  </form>,
]
`;
